src/modules/imageresizer/dll/Settings.cpp

```diff
@@ -1,66 +1,116 @@
 #include "pch.h"
-#include <commctrl.h>
 #include "Settings.h"
 
-const wchar_t c_rootRegPath[] = L"Software\\Microsoft\\ImageResizer";
-const wchar_t c_enabled[] = L"Enabled";
-const bool c_enabledDefault = true;
+#include <common/json.h>
+#include <common/settings_helpers.h>
+#include <filesystem>
+#include <commctrl.h>
 
-bool CSettings::GetEnabled()
+namespace
 {
-    return GetRegBoolValue(c_enabled, c_enabledDefault);
-}
+    const wchar_t c_imageResizerDataFilePath[] = L"\\image-resizer-settings.json";
+    const wchar_t c_rootRegPath[] = L"Software\\Microsoft\\ImageResizer";
+    const wchar_t c_enabled[] = L"Enabled";
 
-bool CSettings::SetEnabled(_In_ bool enabled)
-{
-    return SetRegBoolValue(c_enabled, enabled);
+    unsigned int GetRegNumber(const std::wstring& valueName, unsigned int defaultValue)

```

Comment:
`GetRegNumber` could be misleading name. Maybe `GetRegInteger` would be better

Reply:
You’re right. Or maybe RegReadInteger, RegReadString etc.

Reply:
RegReadInteger sounds good to me

Reply:
Done.